### PR TITLE
Add unit test and e2e test for ResourceDistribution controller

### PR DESCRIPTION
**Pull request by veophi**
### Ⅰ. Describe what this PR does
Add unit test and e2e test for ResourceDistribution controller

### CODE REVIEW

File: test/e2e/apps/resourcedistribution.go [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+
+				// after update targets
+				resourceDistribution.Spec.Targets.IncludedNamespaces.List = []appsv1alpha1.ResourceDistributionNamespace{
+					{Name: nsPrefix4 + "-2"},
+				}
+				resourceDistribution.Spec.Targets.NamespaceLabelSelector.MatchLabels = map[string]string{
+					"e2e-rd-group": "two",
+				}
+				tester.UpdateResourceDistribution(resourceDistribution)
+				matchedNamespaces, unmatchedNamespaces, err = tester.GetNamespaceForDistributor(&resourceDistribution.Spec.Targets)
```
[the remark this answers is not in the document]
Comment (veophi) [reply test/e2e/apps/resourcedistribution.go:316]: fixed

[the remark this answers is not in the document]
Comment (veophi) [reply test/e2e/apps/resourcedistribution.go:316]: ignore

File: pkg/controller/resourcedistribution/resourcedistribution_namespace_event_handler_test.go [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+import (
+	"context"
+	"testing"
+
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/util/workqueue"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
```
[the remark this answers is not in the document]
Comment (veophi) [reply pkg/controller/resourcedistribution/resourcedistribution_namespace_event_handler_test.go:28]: ignore

File: pkg/controller/resourcedistribution/resourcedistribution_controller_test.go [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+	appsv1alpha1 "github.com/openkruise/kruise/apis/apps/v1alpha1"
+	utils "github.com/openkruise/kruise/pkg/webhook/resourcedistribution/validating"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
```
[the remark this answers is not in the document]
Comment (veophi) [reply pkg/controller/resourcedistribution/resourcedistribution_controller_test.go:32]: ignore

[the remark this answers is not in the document]
Comment (veophi) [reply pkg/controller/resourcedistribution/resourcedistribution_controller_test.go:32]: ignore

File: pkg/controller/resourcedistribution/resourcedistribution_namespace_event_handler_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (veophi) [reply pkg/controller/resourcedistribution/resourcedistribution_namespace_event_handler_test.go:28]: ignore

**Review by FillZpp — APPROVED** [review]
/lgtm

**Pull request merged**